vm/parser.ts

```diff
@@ -711,6 +861,17 @@ module J2ME {
       }
     }
 
+    getClassNameSlow(): string {
+      return this._name || (this._name = ByteStream.readString(this.utf8Name));
+    }
```

Comment:
> But bump those numbers, and your version starts slowing down again.

Part of the issue here is that `_x` is of a different type than the final value, setting a sentinel value of the same type as the final value makes the getXXX pattern pull out ahead in all configurations: http://jsfiddle.net/dt4z21af/5/

The memoizer version falls behind here because it allocates a lot of temporary objects. Here is a workaround for that: http://jsfiddle.net/dt4z21af/6/. This closes the gap but falls behind when the number of accesses is low and the number of objects is high.

```
Objects: 128 x Accesses: 32768

Memoizer: 7.31
Call: 7.52

Objects: 256 x Accesses: 16384

Memoizer: 6.38
Call: 6.88

Objects: 512 x Accesses: 8192

Memoizer: 6.61
Call: 6.67

Objects: 1024 x Accesses: 4096

Memoizer: 7.38
Call: 5.64

Objects: 2048 x Accesses: 2048

Memoizer: 7.61
Call: 6.92

Objects: 4096 x Accesses: 1024

Memoizer: 8.41
Call: 5.70

Objects: 8192 x Accesses: 512

Memoizer: 11.04
Call: 6.58

Objects: 16384 x Accesses: 256

Memoizer: 16.19
Call: 5.89

Objects: 32768 x Accesses: 128

Memoizer: 27.84
Call: 7.18
```